packages/forms/src/UIForm/merge.js

```diff
@@ -0,0 +1,207 @@
+import React, { PropTypes } from 'react';
+import { merge } from 'talend-json-schema-form-core';
+import get from 'lodash/get';
+
+import FieldTemplate from './fields/FieldTemplate';
+
+/* eslint-disable no-param-reassign */
+
+function parseArray(items, key) {
+	if (items.type === 'object') {
+		return parseProperties(items.properties, false, `${key}[]`);
+	}
+	return [getUISchemaFromObject(items, key)];
+}
+
+function getUISchemaFromObject(schema, key) {
+	const ui = { key };
```

Comment:
This method is called with `key = undefined` when node to parse is root, this will produce `{ undefined: undefined }`